samples/local_attestation/common/attestation.cpp

```diff
@@ -21,36 +28,43 @@ bool Attestation::generate_local_report(
     const uint8_t* data,
     const size_t data_size,
     uint8_t** report_buf,
-    size_t* remote_report_buf_size)
+    size_t* local_report_buf_size)
 {
     bool ret = false;
     uint8_t sha256[32];
-    oe_result_t result = OE_OK;
-    uint8_t* temp_buf = NULL;
+    oe_result_t result = OE_OK;    
+    oe_result_t attester_result = OE_OK;
+    oe_result_t attester_format_result = OE_OK;
 
     if (m_crypto->Sha256(data, data_size, sha256) != 0)
     {
         goto exit;
     }
 
-    // To generate a local report that just needs to be attested by another
-    // enclave running on the same platform, set flags to 0 in oe_get_report
-    // call. This uses the EREPORT instruction to generate this enclave's local
-    // report.
-    result = oe_get_report(
-        0,      // get a local report
-        sha256, // Store sha256 in report_data field
-        sizeof(sha256),
-        target_info_buffer,
-        target_info_size,
-        &temp_buf,
-        remote_report_buf_size);
+    // Initialize attester and use the SGX plugin.
+    attester_result = oe_attester_initialize();
+    if (attester_result != OE_OK)
+    {
+        TRACE_ENCLAVE("oe_attester_initialize failed.");
+        goto exit;
+    }
+
+    // Select the attestation format.
```

Comment:
oe_attester_select_format() helps you select a format from an array of format. If you just want to generate one specific format, you don't have to call this. Just call:
oe_get_evidence(&sgx_local_uuid, ...)

Reply:
I did it this way because this is the method shown in the tests.  When I try your suggestion, it seems to cause oe_verify_evidence to throw a OE_NOT_FOUND.

Reply:
Never mind, I fixed this issue.